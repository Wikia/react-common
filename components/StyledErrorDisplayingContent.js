'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var styled = _interopDefault(require('styled-components'));

function _taggedTemplateLiteral(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  return Object.freeze(Object.defineProperties(strings, {
    raw: {
      value: Object.freeze(raw)
    }
  }));
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    align-content: center;\n    color: red;\n    display: flex;\n    margin-top: 0;\n    margin-bottom: 0;\n    padding: 10px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    background: #fff;\n    padding: 10px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Wrapper = styled.div(_templateObject());
var Message = styled.div(_templateObject2());
/**
 * A standard error message when content cannot be displayed.
 */

function StyledErrorDisplayingContent() {
  return React.createElement(Wrapper, null, React.createElement(Message, null, "Sorry, we\u2019re having trouble displaying this content right now."));
}

module.exports = StyledErrorDisplayingContent;
