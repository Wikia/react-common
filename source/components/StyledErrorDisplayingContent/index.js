import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
    background: #fff;
    padding: 10px;
`;

const Message = styled.div`
    align-content: center;
    color: red;
    display: flex;
    margin-top: 0;
    margin-bottom: 0;
    padding: 10px;
`;

/**
 * A standard error message when content cannot be displayed.
 */
function StyledErrorDisplayingContent() {
    return (
        <Wrapper>
            <Message>
                Sorry, weâ€™re having trouble displaying this content right now.
            </Message>
        </Wrapper>
    );
}

export default StyledErrorDisplayingContent;
