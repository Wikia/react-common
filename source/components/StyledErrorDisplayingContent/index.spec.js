import { mount } from 'enzyme';
import React from 'react';

import StyledErrorDisplayingContent from './index';

test('StyledErrorDisplayingContent renders', () => {
    const component = mount(
        <StyledErrorDisplayingContent />
    );
    expect(component).toMatchSnapshot();
});
