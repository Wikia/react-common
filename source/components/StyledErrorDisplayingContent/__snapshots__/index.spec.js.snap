// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledErrorDisplayingContent renders 1`] = `
<StyledErrorDisplayingContent>
  <StyledErrorDisplayingContent__Wrapper>
    <div
      className="StyledErrorDisplayingContent__Wrapper-sc-7aqm4x-0 dWUDzc"
    >
      <StyledErrorDisplayingContent__Message>
        <div
          className="StyledErrorDisplayingContent__Message-sc-7aqm4x-1 lkqXoP"
        >
          Sorry, weâ€™re having trouble displaying this content right now.
        </div>
      </StyledErrorDisplayingContent__Message>
    </div>
  </StyledErrorDisplayingContent__Wrapper>
</StyledErrorDisplayingContent>
`;
